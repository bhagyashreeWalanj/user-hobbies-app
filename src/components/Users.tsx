import * as React from 'react'
import { useState, useEffect } from 'react'
import styles from './../styles/Users.module.scss'
import { IUser } from './../interfaces/user'
import UserList from './UserList'

export interface IProps {
  Users: IUser[]
  getAllUsers(): void
  selectedUserId: string
  selectUser(id: string): void
  addUser(id: string): void
}

const Users = (props: IProps) => {
  const [newUserName, setNewUserName] = useState('')
  const { Users, selectedUserId, selectUser, addUser, getAllUsers } = props
  const [error, setError] = useState('')

  const onAddUser = () => {
    if (newUserName !== '') {
      addUser(newUserName)
      setNewUserName('')
    } else {
      setError('Enter user name')
      return
    }
  }

  useEffect(() => {
    getAllUsers()
  }, [getAllUsers])
  return (
    <div>
      <div className={styles.errorDiv}>{error}</div>
      <div className={styles.addUserContainer}>
        <input
          type="text"
          id="newUser"
          placeholder="Enter user name"
          value={newUserName}
          onChange={({
            target: { value },
          }: React.ChangeEvent<HTMLInputElement>) => setNewUserName(value)}
          className={styles.searchContainerField}
        />
        <button
          className={styles.addUserbutton}
          onClick={onAddUser}
          id="addNewbutton"
        >
          Add User
        </button>
      </div>

      <UserList
        Users={Users}
        selectedUserId={selectedUserId}
        selectUser={selectUser}
      />
    </div>
  )
}

export default Users
