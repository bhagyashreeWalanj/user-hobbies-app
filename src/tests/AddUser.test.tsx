import React from 'react'
import { mount, shallow, render } from 'enzyme'
import Users, { IProps } from '../components/Users'

describe('Testing Add New User Component', () => {
  const getAllUsers = () => {}
  const selectUser = () => {}
  const addUser = jest.fn()
  const users = [
    {
      id: '1',
      name: 'John Hecker',
      hobbies: [
        {
          id: '2',
          hobby: 'Playing Cricket',
          passion: 'Low',
          year: '1990',
        },
      ],
    },
  ]
  const props: IProps = {
    Users: users,
    getAllUsers: getAllUsers,
    selectedUserId: '',
    selectUser: selectUser,
    addUser: addUser,
  }

  const wrapper = shallow(<Users {...props} />)
  const container = wrapper.find('#newUser')

  it('Renders Users Component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should display error on empty user field click', () => {
    container.simulate('change', { target: { value: '' } })
    wrapper.find('#addNewbutton').simulate('click')
    const errorDiv = wrapper.find('.errorDiv')
    const errorText = errorDiv.text()
    expect(errorText).toEqual('Enter user name')
  })

  it('should add user in database', () => {
    container.simulate('change', { target: { value: 'John' } })
    wrapper.find('#addNewbutton').simulate('click')
    expect(addUser).toBeCalledTimes(1)
  })
})
