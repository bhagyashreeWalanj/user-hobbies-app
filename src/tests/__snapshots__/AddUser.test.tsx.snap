// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Add New User Component Renders Users Component 1`] = `
<div>
  <div
    className="errorDiv"
  />
  <div
    className="addUserContainer"
  >
    <input
      className="searchContainerField"
      id="newUser"
      onChange={[Function]}
      placeholder="Enter user name"
      type="text"
      value=""
    />
    <button
      className="addUserbutton"
      id="addNewbutton"
      onClick={[Function]}
    >
      Add User
    </button>
  </div>
  <UserList
    Users={
      Array [
        Object {
          "hobbies": Array [
            Object {
              "hobby": "Playing Cricket",
              "id": "2",
              "passion": "Low",
              "year": "1990",
            },
          ],
          "id": "1",
          "name": "John Hecker",
        },
      ]
    }
    selectUser={[Function]}
    selectedUserId=""
  />
</div>
`;
